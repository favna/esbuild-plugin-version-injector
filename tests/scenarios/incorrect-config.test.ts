import tsup, { type Options } from 'tsup';
import type { PluginOptions } from '../../dist';
import { assertFileContent, buildAbsolutePath, createTsupConfig } from '../common/util';

function createConfig(pluginOptions?: PluginOptions): Options {
  return createTsupConfig(
    {
      format: ['esm'],
      entry: [buildAbsolutePath('../fixtures/build-in/typescript.mts')]
    },
    pluginOptions
  );
}

describe('Incorrect Config', () => {
  test('GIVEN incorrect filter THEN warns and uses all filter', async () => {
    const warnMock = vi.fn().mockImplementation((text) => text);
    const originalWarn = console.warn;

    console.warn = warnMock;

    const config = createConfig({
      // @ts-ignore testing incorrect config
      filter: 42
    });

    await tsup.build(config);

    await assertFileContent('../fixtures/build-out/tsup/typescript.mjs');
    expect(warnMock).toHaveBeenCalledOnce();

    console.warn = originalWarn;
  });
});
