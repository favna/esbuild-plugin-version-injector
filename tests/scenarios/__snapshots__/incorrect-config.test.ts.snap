// Vitest Snapshot v1

exports[`Incorrect Config > GIVEN incorrect filter THEN warns and uses all filter 1`] = `
"// tests/fixtures/build-in/typescript.mts
var version = \\"1.0.3\\";

export { version };
"
`;
